.github/workflows/build-hpc.yml

```diff
@@ -0,0 +1,109 @@
+name: build-hpc
+
+# Controls when the action will run
+on:
+
+  # Trigger the workflow on all pushes, except on tag creation
+  push:
+    branches:
+    - '**'
```

Comment:
I think you might want to restrict general branch push runs to "main" and "develop". Pull requests (including draft PRs) and manual triggers (below) should suffice for a priori testing if that's needed.

Reply:
Thanks, good point. I agree that PR and main/develop branches should suffice by default - on demand trigger is still available for other branches